Source/CesiumRuntime/Private/CesiumBoundingVolumeComponent.h

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020-2021 CesiumGS, Inc. and Contributors
+
+#pragma once
+
+#include "Components/PrimitiveComponent.h"
+#include "Components/SceneComponent.h"
+#include "CoreMinimal.h"
+#include "PrimitiveSceneProxy.h"
+#include <Cesium3DTilesSelection/BoundingVolume.h>
+#include <Cesium3DTilesSelection/TileOcclusionRendererProxy.h>
+#include <glm/mat4x4.hpp>
+#include <memory>
+#include <optional>
+#include "CesiumBoundingVolumeComponent.generated.h"
+
+class ACesiumGeoreference;
+
+UCLASS()
+class UCesiumBoundingVolumePoolComponent : public USceneComponent {
+  GENERATED_BODY()
+
+public:
+  UCesiumBoundingVolumePoolComponent();
+
+  /**
+   * Updates bounding volume transforms from a new double-precision
+   * transformation from the Cesium world to the Unreal Engine world.
+   *
+   * @param CesiumToUnrealTransform The new transformation.
+   */
+  void UpdateTransformFromCesium(const glm::dmat4& CesiumToUnrealTransform);
+
+  const std::shared_ptr<Cesium3DTilesSelection::TileOcclusionRendererProxyPool>&
+  getPool() {
+    return this->_pPool;
+  }
+
+private:
+  glm::dmat4 _cesiumToUnreal;
+
+  // These are really implementations of the functions in
+  // TileOcclusionRendererProxyPool, but we can't use multiple inheritance with
+  // UObjects. Instead use the CesiumBoundingVolumePool and forward virtual
+  // calls to the implementations.
+  friend class UCesiumBoundingVolumePool;
+
+  Cesium3DTilesSelection::TileOcclusionRendererProxy* createProxy();
+
+  void destroyProxy(Cesium3DTilesSelection::TileOcclusionRendererProxy* pProxy);
+
+  std::shared_ptr<Cesium3DTilesSelection::TileOcclusionRendererProxyPool>
+      _pPool;
+};
+
+class UCesiumBoundingVolumePool
+    : public Cesium3DTilesSelection::TileOcclusionRendererProxyPool {
+public:
+  UCesiumBoundingVolumePool(UCesiumBoundingVolumePoolComponent* pOutter);
+
+protected:
+  Cesium3DTilesSelection::TileOcclusionRendererProxy* createProxy() override;
+
+  void destroyProxy(
+      Cesium3DTilesSelection::TileOcclusionRendererProxy* pProxy) override;
+
+private:
+  UCesiumBoundingVolumePoolComponent* _pOutter;
+};
+
+UCLASS()
+class UCesiumBoundingVolumeComponent
+    : public UPrimitiveComponent,
+      public Cesium3DTilesSelection::TileOcclusionRendererProxy {
```

Comment:
Hm I'm not actually sure, maybe multiple inheritance doesn't always _not_ work with Unreal classes. But I definitely got some weird issues when trying to have `UCesiumBoundingVolumePoolComponent` inherit from both `USceneComponent` as well as `TileOcclusionRenderProxyPool`. It was some sort of confusing compile error in the auto-generated Unreal header code. 

This case of multi-inheritance doesn't seem to cause much issue though right, would it really be worth splitting it up? If so, would the same pattern make sense here where it would become an inner nested class?